tests/ctest/test_record.c

```diff
@@ -0,0 +1,901 @@
+	{
+		printf ("error/module = %s\n", keyString (m));
+		fflush (stdout);
+	}
+	if ((m = keyGetMeta (key, "meta:/error/file")) != NULL)
+	{
+		printf ("error/file = %s\n", keyString (m));
+		fflush (stdout);
+	}
+	if ((m = keyGetMeta (key, "meta:/error/line")) != NULL)
+	{
+		printf ("error/line = %s\n", keyString (m));
+		fflush (stdout);
+	}
+	if ((m = keyGetMeta (key, "meta:/error/mountpoint")) != NULL)
+	{
+		printf ("error/mountpoint = %s\n", keyString (m));
+		fflush (stdout);
+	}
+	if ((m = keyGetMeta (key, "meta:/error/configfile")) != NULL)
+	{
+		printf ("error/configfile = %s\n", keyString (m));
+		fflush (stdout);
+	}
+	if ((m = keyGetMeta (key, "meta:/error/reason")) != NULL)
+	{
+		printf ("error/reason = %s\n", keyString (m));
+		fflush (stdout);
+	}
+
+	Key * metaRoot = keyNew ("meta:/error", KEY_END);
+	ksDel (ksCut (keyMeta (key), metaRoot));
+	keyDel (metaRoot);
+}
+
+static char * appendString (const char * s1, const char * s2)
+{
+	size_t newPathLength = strlen (s1) + strlen (s2) + 1;
+	char * newPath = elektraMalloc (newPathLength * sizeof (char));
+	snprintf (newPath, newPathLength, "%s%s", s1, s2);
+	return newPath;
+}
+
+static KDB * openPrefixedKdbInstance (KeySet * contract, Key * errorKey, const char * prefix)
+{
+	char * bootstrapPath = appendString (prefix, KDB_DB_INIT);
+	elektraBootstrapPathContract (contract, bootstrapPath);
+	KDB * kdb = kdbOpen (contract, errorKey);
+	free (bootstrapPath);
+
+	for (elektraCursor i = 0; i < ksGetSize (kdb->backends); i++)
+	{
+		Key * backendKey = ksAtCursor (kdb->backends, i);
+		BackendData * backendData = (BackendData *) keyValue (backendKey);
+
+		Key * pathKey = ksLookupByName (backendData->definition, "system:/path", 0);
+		if (pathKey == NULL)
+		{
+			continue;
+		}
+
+		const char * oldPath = keyString (pathKey);
+		char * newPath = appendString (prefix, oldPath);
+		keySetString (pathKey, newPath);
+		elektraFree (newPath);
+	}
+
+	return kdb;
+}
+
+static void closePrefixedKdbInstance (KDB * kdb, Key * errorKey, bool deleteFiles)
+{
+	// remove all files
+	if (deleteFiles)
+	{
+		KeySet * tmp = ksNew (0, KS_END);
+		ElektraKdbPhase phase = ELEKTRA_KDB_GET_PHASE_RESOLVER;
+		ksAppendKey (kdb->global, keyNew ("system:/elektra/kdb/backend/phase", KEY_BINARY, KEY_SIZE, sizeof (ElektraKdbPhase),
+						  KEY_VALUE, &phase, KEY_END));
+		Key * pk = keyNew ("/", KEY_END);
+
+		for (elektraCursor i = 0; i < ksGetSize (kdb->backends); i++)
+		{
+			Key * backendKey = ksAtCursor (kdb->backends, i);
+
+			if (keyGetNamespace (backendKey) == KEY_NS_PROC)
+			{
+				// proc:/ backends only run in poststorage
+				continue;
+			}
+
+			keyCopy (pk, backendKey, KEY_CP_NAME);
+			BackendData * backendData = (BackendData *) keyValue (backendKey);
+
+			backendData->backend->kdbGet (backendData->backend, tmp, pk);
+
+			const char * path = keyString (pk);
+			if (path != NULL && strcmp ("", path) != 0)
+			{
+				elektraUnlink (path);
+			}
+		}
+
+		ksDel (tmp);
+		keyDel (pk);
```

Comment:
I think you can just do a `kdbSet()` with an empty `KeySet`. The `resolver` plugin should then remove all files and that would also work for non-file-based backends.

```suggestion
		KeySet * tmp = ksNew (0, KS_END);
		Key * pk = keyNew ("/", KEY_END);
		
		kdbSet (kdb, tmp, pk);

		ksDel (tmp);
		keyDel (pk);
```

Reply:
Great idea!